components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/FeatureVariables.kt

```diff
@@ -106,20 +181,126 @@ interface Variables {
      */
     fun getDrawable(key: String): Drawable? = null
 
+    /**
+     * Uses `getStringList(key: String)` to get a list of strings, then coerces the
+     * strings in the list into Drawables. Values that cannot be coerced are omitted.
+     */
+    fun getDrawableList(key: String): List<Drawable>? = null
+
+    /**
+     * Uses `getStringList(key: String)` to get a list of strings, then coerces the
+     * values into Drawables. Values that cannot be coerced are omitted.
+     */
+    fun getDrawableMap(key: String): Map<String, Drawable>? = null
+
     // Get a child configuration object.
+
+    /**
+     * Gets a nested `JSONObject` value for this key, and creates a new `Variables` object. If
+     * the value at the key is not a JSONObject, then return `null`.
+     */
     fun getVariables(key: String): Variables? = null
+
+    /**
+     * Gets a list value for this key, and transforms all `JSONObject`s in the list into `Variables`.
+     *
+     * If the value isn't a list, then returns `null`. Items in the list that are not `JSONObject`s
+     * are omitted from the final list.
+     */
+    fun getVariablesList(key: String): List<Variables>? = null
+
+    /**
+     * Gets a map value for this key, and transforms all `JSONObject`s that are values into `Variables`.
+     *
+     * If the value isn't a `JSONObject`, then returns `null`. Values in the map that are not `JSONObject`s
+     * are omitted from the final map.
+     */
+    fun getVariablesMap(key: String): Map<String, Variables>? = null
+
     // This may be important when transforming in to a code generated object.
-    fun <T> getVariables(key: String, transform: (Variables) -> T) = getVariables(key)?.let(transform)
+    fun <T> getVariables(key: String, transform: (Variables) -> T?) = getVariables(key)?.let(transform)
```

Comment:
A quick follow-up thought, is this a breaking change? I don't know enough about kotlin, but is it making a type nullable OK for backward compatibility, or do we need to mark this as a breaking change?

Reply:
> is it making a type nullable OK for backward compatibility

I'd expect that it is a breaking change - IIUC it means callers of this will need to change?

Reply:
Yes, I think strictly this would be a breaking change.

Philosophically: If an API, that no-one uses, changes: is it really a breaking change?